runtime/lua/vim/shared.lua

```diff
@@ -1419,4 +1419,59 @@ function vim._ensure_list(x)
   return { x }
 end
 
+---@param tabpage integer
+---@param layout table
+function vim._set_layout(tabpage, layout)
```

Comment:
Should live in `runtime/lua/vim/_editor.lua`

(`shared.lua` is a special-case for sharing code with the test runner.)